host/tests/rmfAudio_L3_TestCases/rmfAudio_test01_primaryDataCapture.py

```diff
@@ -0,0 +1,211 @@
+#!/usr/bin/env python3
+#** *****************************************************************************
+# *
+# * If not stated otherwise in this file or this component's LICENSE file the
+# * following copyright and licenses apply:
+# *
+# * Copyright 2024 RDK Management
+# *
+# * Licensed under the Apache License, Version 2.0 (the "License");
+# * you may not use this file except in compliance with the License.
+# * You may obtain a copy of the License at
+# *
+# *
+# http://www.apache.org/licenses/LICENSE-2.0
+# *
+# * Unless required by applicable law or agreed to in writing, software
+# * distributed under the License is distributed on an "AS IS" BASIS,
+# * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# * See the License for the specific language governing permissions and
+# * limitations under the License.
+# *
+#* ******************************************************************************
+
+import os
+import sys
+import time
+
+dir_path = os.path.dirname(os.path.realpath(__file__))
+sys.path.append(os.path.join(dir_path, "../"))
+
+from rmfAudioClasses.rmfAudio import rmfAudioClass
+from raft.framework.plugins.ut_raft import utHelperClass
+from raft.framework.plugins.ut_raft.configRead import ConfigRead
+from raft.framework.plugins.ut_raft.utUserResponse import utUserResponse
+
+class rmfAudio_test01_primaryDataCapture(utHelperClass):
+    """
+    Test class to open capture Primary Audio Data and verify it with its source.
+
+    This class uses the `rmfAudioClass` to test primary data capture,
+    downloading necessary test assets, setting up RMF Audio Capture
+    and performing verification of captured audio.
+    """
+
+    # Class variables
+    testName  = "test01_primaryDataCapture"
+    testSetupPath = dir_path + "/rmfAudio_L3_testSetup.yml"
+    moduleName = "rmfaudiocapture"
+    rackDevice = "dut"
+
+    def __init__(self):
+        """
+        Initializes the test class with test name, setup configuration, and sessions for the device.
+
+        Args:
+            None
+        """
+        super().__init__(self.testName, '1')
+
+        # Load test setup configuration
+        self.testSetup = ConfigRead(self.testSetupPath, self.moduleName)
+
+        # Open Session for hal test
+        self.hal_session = self.dut.getConsoleSession("ssh_hal_test")
+
+        cmds = self.testSetup.get("assets").get("device").get(self.testName).get("postcmd")
+        if cmds is not None:
+            for cmd in cmds:
+                self.hal_session.write(cmd)
+
+         # Create user response Class
+        self.testUserResponse = utUserResponse()
+
+        # Get path to device profile file
+        self.deviceProfile = dir_path + "/" + self.cpe.get("test").get("profile")
+
+    def testDownloadAssets(self):

```

Comment:
testDownloadAssets,testCleanAssets,... 
most of these functions are used by the all tests so better write a helper class with all these 
ex: 
https://github.com/rdkcentral/rdk-halif-test-device_settings/blob/3.3.0/host/tests/L3_TestCases/dsHdmiIn/dsHdmiInHelperClass.py

Reply:
Added a helper class and updated tests